ServerCore/Pages/Pieces/InternMeta.cshtml.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.Helpers;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Pieces
+{
+    [Authorize(Policy = "PlayerIsOnTeam")]
+    public class InternMetaModel : EventSpecificPageModel
+    {
+        public InternMetaModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager) : base(serverContext, userManager)
+        {
+        }
+
+        public IList<Piece> EarnedPieces { get;set; }
+        
+        public async Task<IActionResult> OnGetAsync(int puzzleId)
+        {
+            if (!Event.IsInternEvent)
```

Comment:
Rather than limiting to intern events, I'd suggest having auth be that puzzleId be unlocked. That would also make module metas or puzzle hunt metas work.

Reply:
I am explicitly limiting to intern events because this approach may be dangerous for nonintern events and I do not want to expose it for general module meta use.

Reply:
That strikes me as more specific than necessary. Could we reasonably let a puzzle opt into this and perhaps have a way to tell which puzzles feed it? (My last module meta could have used this page, for example)

Reply:
Your last module meta could NOT have used this page because this page is counting solves over the entire event.

Reply:
Now supports a event-wide mode that looks at all puzzles and a module mode that looks for matching Group tags. With this change, it's approved for module meta use.